packages/server/src/fhir/transaction.test.ts

```diff
@@ -0,0 +1,270 @@
+import { OperationOutcomeError, Operator } from '@medplum/core';
+import { Patient } from '@medplum/fhirtypes';
+import { randomUUID } from 'crypto';
+import { initAppServices, shutdownApp } from '../app';
+import { loadTestConfig } from '../config';
+import { withTestContext } from '../test.setup';
+import { Repository } from './repo';
+
+describe('FHIR Repo', () => {
+  let repo: Repository;
+
+  beforeAll(async () => {
+    const config = await loadTestConfig();
+    await initAppServices(config);
+
+    const clientApp = 'ClientApplication/' + randomUUID();
+    const projectId = randomUUID();
+    repo = new Repository({
+      extendedMode: true,
+      project: projectId,
+      author: {
+        reference: clientApp,
+      },
+    });
+  });
+
+  afterAll(async () => {
+    await shutdownApp();
+  });
+
+  test('Transaction commit', () =>
+    withTestContext(async () => {
+      let patient: Patient | undefined;
+      await repo.withTransaction(async () => {
+        patient = await repo.createResource<Patient>({ resourceType: 'Patient' });
+        expect(patient).toBeDefined();
+      });
+      expect(patient).toBeDefined();
+
+      // Read the patient by ID
+      // This should succeed
+      const readCheck1 = await repo.readResource('Patient', patient?.id as string);
+      expect(readCheck1).toBeDefined();
+
+      // Search for patient by ID
+      // This should succeed
+      const searchCheck1 = await repo.search<Patient>({
+        resourceType: 'Patient',
+        filters: [{ code: '_id', operator: Operator.EQUALS, value: patient?.id as string }],
+      });
+      expect(searchCheck1).toBeDefined();
+      expect(searchCheck1.entry).toHaveLength(1);
```

Comment:
These could optionally be combined, up to your preference
```suggestion
      expect(searchCheck1?.entry).toHaveLength(1);
```